modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/processor/PortletRegistryImpl.java

```diff
@@ -127,12 +118,22 @@ public List<String> getFragmentEntryLinkPortletIds(
 
 	@Override
 	public List<String> getPortletAliases() {
-		return new ArrayList<>(_portletNames.keySet());
+		return new ArrayList<>(_aliasPortletNames.keySet());
 	}
 
 	@Override
 	public String getPortletName(String alias) {
-		return _portletNames.get(alias);
+		return _aliasPortletNames.get(alias);
+	}
+
+	@Override
+	public void registerAlias(String alias, String portletName) {
+		_aliasPortletNames.put(alias, portletName);
+	}
+
+	@Override
+	public void unregisterAlias(String alias, String portletName) {
+		_aliasPortletNames.remove(alias, portletName);
```

Comment:
@W2jx Please do not do `remove(alias, portletName)`, do `remove(alias)` directly, I did not see any reason you have to do this.